understanding/sharding.md

```diff
@@ -11,20 +11,168 @@ You do not need to worry about sharding until your bot hits around 2,400 guilds.
 There are additional difficulties when sharding a bot that add complexity to your code \(one of the reasons you shouldn't shard too early\).
 
 * Collections do not cache data from all shards, so you can't grab data from a guild in another shard easily.
-* In order to do anything across shards you need to worry about using `broadcastEval` and such \(tutorial coming soon!\).
+* In order to do anything across shards you need to worry about using `broadcastEval` and such \(Examples and explanation below\).
 * Sharded bots often gain very marginal performance increase and might even use _more_ memory due to using more node processes.
 * If you're using any sort of database or connection, multiple shards may cause issues with multiple processes connecting to a single end point.
 
-## Example Sharding Code
+## Example Sharding Manager Code
 
 ```javascript
 /*
     The following code goes into it's own file, and you run this file
     instead of your main bot file.
 */
 
-const Discord = require('discord.js');
-const Manager = new Discord.ShardingManager('./YOUR_BOT_FILE_NAME.js');
-Manager.spawn(2); // This example will spawn 2 shards (5,000 guilds);
+// Include discord.js ShardingManger
+const { ShardingManager } = require('discord.js');
+
+// Create your ShardingManger instance
+const manager = new ShardingManager('./YOUR_BOT_FILE_NAME.js', {
+    // for ShardingManager options see:
+    // https://discord.js.org/#/docs/main/master/class/ShardingManager
+    totalShards: 'auto',
+    token: 'YOUR_TOKEN_GOES_HERE'
+});
+
+// Spawn your shards
+manager.spawn();
+
+// Emitted when a shard is created
+manager.on('shardCreate', (shard) => console.log(`Shard ${shard.id} launched`));
+```
+
+## Sharing Information Between Shards
+
+{% hint style="info" %}
+Information is not readily available between shards. In order to get or share information across shards, you will need to make use of either `fetchClientValues()` or `broadcastEval()`.
+{% endhint %}
+
+Remember how we were talking about sharding being a method of "splitting" the bot into multiple instances of itself? Because your sharded bot is now in separate, individual instances, things like your adding your total guilds or getting a specific guild are not as simple as they were before. We must now use either [`fetchClientValues`](https://discord.js.org/#/docs/main/master/class/ShardClientUtil?scrollTo=fetchClientValues) or [`broadcastEval`](https://discord.js.org/#/docs/main/master/class/ShardClientUtil?scrollTo=broadcastEval) to get information from across shards.
```

Comment:
Same here:

```suggestion
Remember how we were talking about sharding being a method of "splitting" the bot into multiple instances of itself? Because your sharded bot is now in separate, individual instances, things like your adding your total guilds or getting a specific guild are not as simple as they were before. We must now use either [`fetchClientValues`](https://discord.js.org/#/docs/main/stable/class/ShardClientUtil?scrollTo=fetchClientValues) or [`broadcastEval`](https://discord.js.org/#/docs/main/stable/class/ShardClientUtil?scrollTo=broadcastEval) to get information from across shards.
```

Reply:
Actually, now that I think of it, you made dedicated sections for both methods. You should link to them, but I'm not sure how to do so with Gitbook. ``[`fetchClientValues`](#FetchClientValues)`` perhaps?

Reply:
I'll ask evie and york how to do this and implement it, sure.